### ISSUE TITLE
compare models with and  without weight

### ISSUE THREAD

**Issue by AdolHong** [jdb78/pytorch-forecasting#1669]
sometimes we want to   compare model with weight to without weight, but,   because the val metric is multiple weight,  this multiplication  make  models cannot be compared.

you can change code below:

file: pytorch_forecasting/models/base_model.py , line 665:

    def validation_step(self, batch, batch_idx):
        x, y = batch
        if y[1] is not None:
            weight = torch.where(y[1]!=1.0, torch.tensor(1.0), y[1])
            y = (y[0], weight)
        log, out = self.step(x, y, batch_idx)
        log.update(self.create_log(x, y, out, batch_idx))
        self.validation_step_outputs.append(log)
        return log

**Comment by fkiraly** [comment]
Interesting suggestion - a pull request would be appreciated so we can see the change in the context of the current code.
